### PR TITLE
[Merged by Bors] - chore: remove nonterminal simp

### PR DESCRIPTION
**Pull request by BoltonBailey**
Removes nonterminal simps on lines looking like `simp [...]`

---

- [x] depends on: https://github.com/leanprover/lean4/issues/2131

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by BoltonBailey** [comment]
Lean 4 issue 2131 seems like it would make this easier, since I am currently trying to fix these by hand.

File: Mathlib/Tactic/NormNum/Basic.lean [hunk 1]
```diff
@@ -348,10 +349,10 @@ theorem isRat_mul {α} [Ring α] {f : α → α → α} {a b : α} {na nb nc : 
   have H := (Nat.cast_commute (α := α) da db).invOf_left.invOf_right.right_comm
   have h₁ := congr_arg (Int.cast (R := α)) h₁
   simp only [Int.cast_mul, Int.cast_ofNat] at h₁
-  simp [← mul_assoc, (Nat.cast_commute (α := α) da nb).invOf_left.right_comm, h₁]
+  simp only [<-mul_assoc, (Nat.cast_commute (α := α) da nb).invOf_left.right_comm, h₁]
```
Comment (joelriou) [inline Mathlib/Tactic/NormNum/Basic.lean:352]:
```suggestion
  simp only [← mul_assoc, (Nat.cast_commute (α := α) da nb).invOf_left.right_comm, h₁]
```

Comment (joelriou) [reply Mathlib/Tactic/NormNum/Basic.lean:352]: (Otherwise, the PR looks good to me.)

**Comment by adomani** [comment]
LGTM!  I am sorry to have caused so much pain, but I think that this is a strong improvement!

I agree with the comment that ideally the linter should be implemented in lean directly, but currently, this is what is available!

Thanks!

**Comment by BoltonBailey** [comment]
> LGTM! I am sorry to have caused so much pain, but I think that this is a strong improvement!
> 
> I agree with the comment that ideally the linter should be implemented in lean directly, but currently, this is what is available!
> 
> Thanks!

What doesn't kill you makes you stronger!

**Comment by joelriou** [comment]
Thanks!

bors merge

**Pull request closed without merging**